jerry-core/jerry-snapshot.c

```diff
@@ -564,7 +566,7 @@ jerry_exec_snapshot (const void *snapshot_p, /**< snapshot */
 
   static const char * const invalid_version_error_p = "Invalid snapshot version";
   static const char * const invalid_format_error_p = "Invalid snapshot format";
-  const uint8_t *snapshot_data_p = (uint8_t *) snapshot_p;
+  const uint32_t *snapshot_data_p = snapshot_p;
```

Comment:
In this case I would keep the `const uint8_t *snapshot_data_p` form here. The asserts later are ok.